src/azul/indexer/__init__.py

```diff
@@ -64,6 +68,10 @@ class SourceSpec(ABC, Generic[SOURCE_SPEC]):
 
     prefix: str = ''
 
+    def __attrs_post_init__(self):
+        assert ':' not in self.prefix
```

Comment:
I want you to think about the ordering of these two lines and tell me what might be wrong here.

Reply:
If we change the order:
the presence of a `:` in the prefix will cause `validate_uuid_prefix` to raise an InvalidUUIDPrefix error, and we would never reach the following `assert`. 

If we don't change the order:
The assertion can be used to express the invariant that a prefix does not contain a `:`, if that is asserted, we can inspect if the uuid prefix is valid.

There might be something I'm missing here.